support/Pipelines/EccentricityControl/EccentricityControl.py

```diff
@@ -0,0 +1,419 @@
+    ecc = amplitude / initial_separation / omega
+
+    print(
+        f"==== Function fitted to dOmega/dt: {name:30s},  rms = {rms:4.3g} "
+        " ===="
+    )
+
+    # Plot dD/dt
+    plt.subplot(2, 2, 1)
+    plt.plot(
+        x,
+        F(p, x),
+        style,
+        label=f"{name:s} \n rms = {rms:2.1e}, ecc = {ecc:4.5f}",
+    )
+    plt.legend(loc=(1.1, -1.3))
+
+    # Plot residual
+    plt.subplot(2, 2, 3)
+    plt.plot(x, errfunc(p, x, y), style, label=name)
+    plt.title("Residual")
+
+    # Print fit parameters
+    print("Fit parameters:")
+    if np.size(p) == 3:
+        print(
+            f"Oscillatory part: (B, w)=({p[0]:4.3g}, {p[1]:7.4f}), ",
+            f"Polynomial part: ({p[2]:4.2g})",
+        )
+    else:
+        print(
+            (
+                f"Oscillatory part: (B, w, phi)=({p[0]:4.3g}, {p[1]:6.4f},"
+                f" {p[2]:6.4f}), "
+            ),
+            end="",
+        )
+        if np.size(p) >= 4:
+            print(f"Polynomial part: ({p[3]:4.2g}, ", end="")
+            for q in p[4:]:
+                print(f"{q:4.2g}", end="")
+            print(")")
+
+    # Print eccentricity
+    print(f"Eccentricity based on fit: {ecc:9.6f}")
+    # Print suggested updates based on fit
+    print("Suggested updates based on fit:")
+    print(f"(dOmega, dadot) = ({dOmg:+13.10f}, {dadot:+8.6g})")
+
+    if initial_xcts_values is not None:
+        xcts_omega, xcts_expansion = initial_xcts_values
+        print("Updated Xcts values based on fit:")
+        print(
+            f"(Omega, adot) = ({(xcts_omega + dOmg):13.10f},"
+            f" {(xcts_expansion + dadot):13.10g})"
+        )
+
+    return p
+
+
+@click.command(name="ecc-remove")
+@click.argument(
+    "h5_file",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, readable=True),
+)
+@click.option(
+    "--subfile-name-aha",
+    "-A",
+    help=(
+        "Name of subfile containing the apparent horizon centers for object A."
+    ),
+)
+@click.option(
+    "--subfile-name-ahb",
+    "-B",
+    help=(
+        "Name of subfile containing the apparent horizon centers for object B."
+    ),
+)
+@click.option(
+    "--tmin",
+    type=float,
+    nargs=1,
+    help=(
+        "The lower time bound to start the fit. Used to remove initial junk and"
+        "transients in the coordinate separations."
+    ),
+)
+@click.option(
+    "--tmax",
+    type=float,
+    nargs=1,
+    help=(
+        "The upper time bound to start the fit. A reasonable value would"
+        " include 2-3 orbits."
+    ),
+)
+@click.option(
+    "--angular-velocity-from-xcts",
+    type=float,
+    nargs=1,
+    help="Value of the angular velocity used in the Xcts file.",
+)
+@click.option(
+    "--expansion-from-xcts",
+    type=float,
+    nargs=1,
+    help="Value of the expansion velocity (adot) used in the Xcts file.",
+)
+@apply_stylesheet_command()
+@show_or_save_plot_command()
+def eccentricity_control_command(
```

Comment:
I can give it a try. I could easily do (2) and (3), I think. For (1) maybe I could collect the updates from all models and output that --since there is no single update here.

Reply:
Sounds good. It's fine to return any or all of the possible updates, then we can use it already to build the automation pipeline and work on improving the fits later or at the same time.